### PR TITLE
Tpetra: Adding tests to resolve #1408

### LINKED ISSUE #1408: Tpetra: Add additional tests for matrix pack/unpack
Tests should exercise the case where the number of rows to pack is not the same as number of rows in matrix.

See #1374 for motivation and details.  Specifically, this [comment](https://github.com/trilinos/Trilinos/issues/1374#issuecomment-307107437) by @maherou  

@trilinos/tpetra

### PR DESCRIPTION
**Pull request by tjfulle**
Build/Test Case Summary
Enabled Packages: TpetraCore
Disabled Packages: PyTrilinos,Claps,TriKota
Enabled all Forward Packages

Build test results:
-------------------
0) MPI_RELEASE_DEBUG_SHARED_PT => passed: passed=1534,notpassed=0 (359.42 min)

### CODE REVIEW

**Comment by tjfulle** [comment]
@trilinos/tpetra @mhoemmen

File: packages/tpetra/core/test/CrsMatrix/CrsMatrix_PackUnpack.cpp [hunk 1]
```diff
@@ -539,9 +542,214 @@ TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(CrsMatrix, PackError, SC, LO, GO, NT)
...
+  RCP<const Comm<int> > comm =
+    Tpetra::DefaultPlatform::getDefaultPlatform ().getComm ();
+  const int world_size = comm->getSize();
+  const int world_rank = comm->getRank();
+
+  // MPI_Comm_size returns the same value on all processes in the
+  // communicator, so we don't need to do a test before exiting early.
+  TEST_ASSERT( world_size % 2 == 0 );
+  if (world_size % 2 != 0) {
+    out << "This test must be run with multiples of 2 MPI processes, but was "
```
Comment (mhoemmen) [inline packages/tpetra/core/test/CrsMatrix/CrsMatrix_PackUnpack.cpp]: It doesn't look like this test depends on MPI at all.  You're just packing data and unpacking it.  The test should work just fine with any number >= 1 of MPI processes.

**Comment by tjfulle** [comment]
@mhoemmen , no, it shouldn't depend on MPI.  This is a case of copy paste a test that was itself copy and pasted from another test that was itself...

I didn't know about running different tests with command line flags.  As is, the file has three tests and each run once.  But, each test is listed separately in CmakeList.txt.  I'll look for an example of running multiple tests in a single file.

**Comment by mhoemmen** [comment]
@tjfulle I would actually prefer that new tests go into new files, rather than having one giant file that runs all the tests.  Giant files take longer to compile ;-) .

**Pull request closed without merging**